archive/imo/imo1975_q1.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2022 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+import data.real.basic
+import data.nat.interval
+import algebra.order.rearrangement
+import algebra.big_operators.ring
+
+/-!
+# IMO 1975 Q1
+
+Let `x₁, x₂, ... , xₙ` and `y₁, y₂, ... , yₙ` be two sequences of real numbers, such that
+`x₁ ≥ x₂ ≥ ... ≥ xₙ` and `y₁ ≥ y₂ ≥ ... ≥ yₙ`. Prove that if `z₁, z₂, ... , zₙ` is any permutation
+of `y₁, y₂, ... , yₙ`, then
+`∑ i in finset.Icc 1 n, (x i - y i) ^ 2 ≤ ∑ i in finset.Icc 1 n, (x i - z i) ^ 2`
```

Comment:
Sometimes I think it's nice to take a bit of licence with doc strings for the sake of readability.

For example here we could write:
```suggestion
`∑ (xᵢ - yᵢ)^2 ≤ ∑ (xᵢ - zᵢ)^2`
```

Reply:
(I guess if you do opt for this, then the same remarks apply to the informal "Solution" section)

Reply:
(I guess if you do opt for this, then the same remarks apply to the informal "Solution" section)